crypto/s2n_sequence.c

```diff
@@ -25,7 +25,8 @@
 
 int s2n_increment_sequence_number(struct s2n_blob *sequence_number)
 {
-    for (int i = sequence_number->size - 1; i >= 0; i--) {
+    for (uint32_t j = sequence_number->size; j > 0; j--) {
+        uint32_t i = j - 1;
```

Comment:
@camshaft should we keep it like that or use `uint64_t`?

Reply:
Same comment as on `s2n_drbg.c`: `int64_t` in place of `uint64_t`.